wpeverest/user-registration
# UR-1656 Tweak - Catch mail send error

### All Submissions:






### Changes proposed in this Pull Request:
Previously when our User Registration Plugin's mail was not being sent or received by end-users, there was no way to know what was actually the issue. This PR introduces a notice displayed to the admin when 5 consecutive email send is failed with a detailed error message or report which will be displayed in our User Registration -> Tools under `ur_email_log` Log.

Also, the Test Email section has been moved to the top of User Registration -> Settings -> Emails section so that when user visits the page, he would be encouraged to check if mail is being sent or not, if not then same detailed error message will be displayed to the user.

### How to test the changes in this Pull Request:

1. It is hard to replicate such issues as email will be sent in our local environment.
2. To test this, install any smtp plugin and enable a mailer and configure it to fail. For example, Install WP MAIL SMTP plugin, setup google as mailer and do not configure it fully. A user authentication related error occur while sending a mail from the system.
3. Click on Send Test Email, a detailed error message should be displayed and log should have been added to User Registration -> Tools section. Verify it.
4. Now try to register multiple users, if the mail send failed threshold crosses 5, then a notice will be displayed to the admin in admin area.

### Types of changes:




### Other information:




### Changelog entry

> Tweak - Catch mail send error.
